pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -563,6 +569,55 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     return brokerResponse;
   }
 
+  private void handleTimestampIndexOverride(PinotQuery pinotQuery, TableConfig tableConfig) {
+    if (tableConfig != null && tableConfig.getFieldConfigList() == null) {
+      return;
+    }
+    Set<String> timestampIndexColumns = tableConfig.getFieldConfigList().stream()
+        .filter(fieldConfig -> fieldConfig.getIndexTypes().contains(FieldConfig.IndexType.TIMESTAMP))
+        .map(fieldConfig -> fieldConfig.getName()).collect(Collectors.toSet());
+    for (Expression expression : pinotQuery.getSelectList()) {
+      setTimestampIndexExpressionOverrideHints(expression, timestampIndexColumns, pinotQuery);
+    }
+    if (pinotQuery.getFilterExpression() != null) {
+      setTimestampIndexExpressionOverrideHints(pinotQuery.getFilterExpression(), timestampIndexColumns, pinotQuery);
+    }
+    if (pinotQuery.getGroupByList() != null) {
+      pinotQuery.getGroupByList().stream().forEach(
```

Comment:
When there's no filter don't stream, just go straight to forEach, I think there's an IDE warning for this in IDEA

Reply:
done.